### DISCUSSION TITLE
recorded ts video doesnt good

### DISCUSSION THREAD

**Discussion by mkarabag** [bluenviron/gortsplib#309] [Questions]
Hi when i use example app which named "client-read-format-h264-save-to-disk". When i play the recorded video. The video is not good i mean sometimes frames corrupted like full green etc.
![Screenshot from 2023-05-31 23-10-31](https://github.com/bluenviron/gortsplib/assets/13116030/26f93ac1-dd3b-4fc9-b107-286529ba282d)

**Comment by aler9** [comment]
Hello, corrupted frames means that the bandwidth between the camera and the client is not always enough, and some frames get lost in transit. You can detect when this happens by using the `OnDecodeError` callback inside `Client`:

```go
c := gortsplib.Client{
    OnDecodeError: func(err error) {
        fmt.Println("decode error:", err)
    },
}
```

In order to mitigate the issue, you can

* use the TCP transport protocol:

   ```go
   t := gortsplib.TransportTCP
   c := gortsplib.Client{
       Transport: &t,
   }
   ```

* decrease the camera bitrate

* increase the bandwidth between the camera and the server by freeing resources, changing routers or cables, etc.
